backend/tests/v5/unit/models/fileRefs.test.js

```diff
@@ -133,7 +145,65 @@ const testRemoveAllFilesFromModel = () => {
 	});
 };
 
+const testDownloadFiles = () => {
+	describe('Download files', () => {
+		test('should throw error if the revision has no files', async () => {
+			const revision = { rFile: [] };
+			await expect(FileRefs.downloadFiles('someTS', 'someModel', revision)).rejects.toEqual(templates.noFileFound);
+		});
+
+		test('should throw error if the revision has no entry', async () => {
+			const revision = { rFile: ['b74ba13b-71db-4fcc-9ff8-7f640aa3dec2P2006823-BDP-XX-A-Stanmore-Envelope_optimized_ifc'] };
+			jest.spyOn(db, 'findOne').mockResolvedValue(undefined);
+			await expect(FileRefs.downloadFiles('someTS', 'someModel', revision)).rejects.toEqual(templates.noFileFound);
+		});
+
+		test('should download the revision files using grid fs if entry has invalid type', async () => {
+			const filename = 'b74ba13b-71db-4fcc-9ff8-7f640aa3dec2P2006823-BDP-XX-A-Stanmore-Envelope_optimized_ifc';
+			const revision = { rFile: ['b74ba13b-71db-4fcc-9ff8-7f640aa3dec2P2006823-BDP-XX-A-Stanmore-Envelope_optimized_ifc'] };
+			const fileEntry = { size: '12345', type: unrecognisedType };
+			const expectedData = { filename: 'P2006823-BDP-XX-A-Stanmore-Envelope_optimized.ifc', readStream: undefined, size: '12345' };
+			const fn = jest.spyOn(db, 'findOne').mockResolvedValue(fileEntry);
+			const res = await FileRefs.downloadFiles('someTS', 'someModel', revision);
+			expect(res).toEqual(expectedData);
+			expect(fn.mock.calls.length).toBe(1);
+			expect(fn.mock.calls[0][0]).toEqual('someTS');
+			expect(fn.mock.calls[0][1]).toEqual('someModel.history.ref');
+			expect(fn.mock.calls[0][2]).toEqual({ _id: filename });
+		});
+
+		test('should download the revision files', async () => {
+			const filename = 'b74ba13b-71db-4fcc-9ff8-7f640aa3dec2P2006823-BDP-XX-A-Stanmore-Envelope_optimized_ifc';
```

Comment:
too ad-hoc, just use generateUUIDString() and generateRandomString() to make up this name.